### PR TITLE
Refactor cross container and bump runwasi

### PR DESCRIPTION
**Pull request by jprendes**
This PR refactors and unifies the `Dockerfile`s used by `cross-rs` for the different shims.
As part of the refactor:
* the arm neon SIMD polyfill is moved upstream.
* `runwasi` is bumped to latest commit
* `libseccomp` is made available in the container, as it's not required by `runwasi`

We also accidentally lost cgroups v1 support with #148. This PR re-enables it by bumping `runwasi` (which now enables it unconditionally).

**Note:**
~~There are failures running the shims when seccomp is enabled.~~ -> fixed by bumping runwasi to include youki's UB fix.

### CODE REVIEW

**Review by Mossaka — APPROVED** [review]
lgtm!

File: cross/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+ARG CROSS_BASE_IMAGE
+FROM $CROSS_BASE_IMAGE
+
+COPY --from=jorgeprendes420/apk-anywhere / /
```
Comment (Mossaka) [inline cross/Dockerfile:4]: Could you please comment on what this is doing?

**Comment by jprendes** [comment]
rebased

**Comment by jprendes** [comment]
It looks like enabling seccomp is messing with the CI tests.
The built shim does run properly when using `docker`, but fails when running with `k3d` in CI.

**Comment by jprendes** [comment]
> It looks like enabling seccomp is messing with the CI tests.

It turns out it was a bug in upstream `runwasi`: https://github.com/containerd/runwasi/pull/364

Everything should work fine with `libseccomp` now.

**Comment by Mossaka** [comment]
Thanks, @jprendes. Merging it now.

**Pull request merged**